### PR TITLE
TLS Support

### PR DESCRIPTION
**Pull request by zonito**
Guide me for Unit Test, if it is a must!

### CODE REVIEW

**Comment by tagomoris** [comment]
I should have to say "must!", but I know it's not so easy.
@zonito Are you trying this code in your environment?

**Comment by tagomoris** [comment]
Just note: https://github.com/fluent/fluent-logger-golang/pull/37

**Comment by zonito** [comment]
@gFazzari Yes!

@tagomoris Yes, I am running in our load test.

File: fluent/fluent.go [hunk 1]
```diff
@@ -418,6 +419,13 @@ func (f *Fluent) connect(ctx context.Context) (err error) {
 		f.conn, err = f.dialer.DialContext(ctx,
 			f.Config.FluentNetwork,
 			f.Config.FluentHost+":"+strconv.Itoa(f.Config.FluentPort))
+	case "tls":
+		config := &tls.Config{InsecureSkipVerify: true}
```
Comment (tagomoris) [inline fluent/fluent.go:423]: This fixed value can't be enabled for everyone.
Could you add a config parameter to specify `InsecureSkipVerify` and set `false` in default?

Comment (tagomoris) [reply fluent/fluent.go:423]: And the variable name `config` is too generic. Could you use `tlsConfig` or something else here?

**Review by tagomoris — CHANGES_REQUESTED** [review]
This change looks generally good to me.
I added two comments about details, and I want one more update [on README about FluentNetwork](https://github.com/fluent/fluent-logger-golang#fluentnetwork) to add "tls" as an option there.

File: fluent/fluent.go (hunk 1 above)
Comment (zonito) [reply fluent/fluent.go]: Done

**Comment by tagomoris** [comment]
@zonito DCO check is now failing. Could you check the instruction?

File: fluent/fluent.go (hunk 1 above)
Comment (tagomoris) [reply fluent/fluent.go]: @zonito Oh, no, I meant that:
* we should have a parameter in [Config](https://github.com/fluent/fluent-logger-golang/blob/master/fluent/fluent.go#L45-L72) about `InsecureSkipVerify` (named probably as `TlsInsecureSkipVerify`)
* and its default value should be `false`
* then use that value here

Comment (tagomoris) [reply fluent/fluent.go]: IMO, we should be able to skip the verification (by specifying `InsecureSkipVerify: true`), but using `true` in default is too dangerous. Users must specify it by themselves by their own intention.

**Comment by zonito** [comment]
@tagomoris Please confirm. I have also tried to address DCO check

File: fluent/fluent.go [hunk 2]
```diff
@@ -36,6 +37,9 @@ const (
 	// Default sub-second precision value to false since it is only compatible
 	// with fluentd versions v0.14 and above.
 	defaultSubSecondPrecision = false
+
+	// Default TLS : Disabled
```
Comment (tagomoris) [inline fluent/fluent.go:41]:
```suggestion
	// Default value whether to skip checking insecure certs on TLS connections
```

File: fluent/fluent.go [hunk 3]
```diff
@@ -69,6 +73,9 @@ type Config struct {
 	// respond with an acknowledgement. This option improves the reliability
 	// of the message transmission.
 	RequestAck bool `json:"request_ack"`
+
+	// To enable TLS
```
Comment (tagomoris) [inline fluent/fluent.go:77]:
```suggestion
	// Flag to skip verifying insecure certs on TLS connections
```

**Review by tagomoris — COMMENTED** [review]
@fluent/team-golang Do you have any idea, guys?

**Comment by tagomoris** [comment]
We can't merge comments without `--signoff` option.

**Comment by zonito** [comment]
Closing this duplicate PR because of #107

**Pull request closed without merging**